configservice/config-service.smithy

```diff
@@ -0,0 +1,62 @@
+// config-service.smithy
+// Declaration of data types used for information exchange between a wasmCloud host
+// and a wasmCloud configuration service
+
+// Tell the code generator how to reference symbols defined in this namespace
+metadata package = [{
+    namespace: "org.wasmcloud.interface.configservice",
+    crate: "wasmcloud_interface_configservice"    
+}]
+
+namespace org.wasmcloud.interface.configservice
+
+/// A request made by a wasmCloud host at startup that contains the host
+/// profile, like its labels. These labels are used by the configuration service
+/// to decide which configuration profile to include in the reply
+structure ConfigurationRequest {
+    @required
+    labels: LabelMap
+}
+
+map LabelMap {
+    key: String,
+    value: String,
+}
+
+/// Contains the configuration profile as deemed appropriate by the configuration
+/// service. A host should react accordingly to this profile upon startup and during
+/// real-time configuration change notifications on the "push" topic. It's noteworthy
+/// that this configuration profile does NOT contain any connection information for NATS,
+/// as allowing that to change at runtime would be a security and reliability risk
+structure HostConfigurationProfile {
+    /// The list of capability providers a host should automatically start
+    autoStartProviders: ProviderReferenceList
+
+    /// A set of credentials the host can use for fetching artifacts
+    registryCredentials: RegistryCredentialMap
+}
```

Comment:
Not sure if this is just the starting point, but for a configuration profile I'd love to be able to start actors, providers, and establish link definitions